src/main/scala/tech/cryptonomic/conseil/generic/chain/RemoteRpc.scala

```diff
@@ -0,0 +1,78 @@
+package tech.cryptonomic.conseil.generic.chain
+
+trait RemoteRpc[Eff[_], Req[_], Res[_]] {
```

Comment:
The naming is totally free to change. The trait corresponds and replaces the existing RPCInterface.
About Req, Res, they are there to accomodate for multi-data requests, like
- sending a batch of inputs (e.g. a List[Offset])
- returning a result value (a representation of the REST content, e.g. json) wrapped in some extra type, to allow a result that contains the request argument. An example is the type `(Offset, Json)`, so that `Res[_] = (Offset, _)`, which has a single type whole to contain the actual result type.

Instead the `Future/IO` effect is represented by the `Eff[_]` type

Reply:
I think it may be useful for our codebase, whenever we use an abstract effect type, to list 2-3 concrete examples in the documentation, as it may be non obvious to newcomers to the codebase what we're generalizing over.